fancy_collections/formatting.py

```diff
@@ -45,25 +47,49 @@ def key_to_string(self, key: Any, obj: Any) -> str:  # noqa
     def to_string(self) -> str:
         if len(self._obj) == 0:
             return self._stringify_empty_class(self._obj)
+
+        objects = {}
+        widths = {}
         for key, val in self._obj.items():
             key = self.key_to_string(key, val)
-            try:
-                string = self.stringify(val)
-            except Exception:  # noqa
-                string = NotImplemented
-            if string is NotImplemented:
-                string = str(val)
-            self._add(key, string.splitlines())
+            lines = self.stringify(val).splitlines()
+            objects[key] = lines
+            widths[key] = self._get_maxlen(lines + [key])
+
+        display_width = int(shutil.get_terminal_size().columns * 0.8)
+
+        keys = tuple(widths.keys())
+        front_keys, back_keys = set(), set()
+        line_length = 0
+        for fkey, bkey in zip(keys, keys[::-1]):
+            line_length += widths[fkey]
+            if line_length < display_width:
+                front_keys.add(fkey)
+            line_length += widths[bkey]
+            if line_length < display_width:
+                back_keys.add(bkey)
+
+        for k, v in objects.items():
+            if k in front_keys:
+                self._add(k, v)
+
+        if not front_keys.intersection(back_keys):
+            self._add("...", ["..."])
+
+        for k, v in objects.items():
+            if k in back_keys and k not in front_keys:
+                self._add(k, objects[k])
+
         return self._render()
 
-    def stringify(self, obj: Any) -> str | NotImplemented:
+    def stringify(self, obj: Any) -> str:
         if isinstance(obj, pd.Index):
             return self._stringify_Index(obj)
         if isinstance(obj, pd.Series):
             return self._stringify_Series(obj)
         if isinstance(obj, pd.DataFrame):
             return self._stringify_DataFrame(obj)
-        return NotImplemented
+        return str(obj)
```

Comment:
The `NotImplemented` feature is for classes that extend this one.  the inherit class can simply super call this method, and just implement the new objects stringification. 

```py
def stringify(self, obj):
    string = super().stringify(obj)
    if string is NotImplemented:
        string = ... # do your custom magic
    return string
```

Reply:
So, you want it back?

Reply:
ok keep it